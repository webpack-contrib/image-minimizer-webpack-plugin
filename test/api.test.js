import ImageMinimizerPlugin from "../src/index";

describe("api", () => {
  describe("basic", () => {
    it("should exported", () => {
      expect(ImageMinimizerPlugin).toBeInstanceOf(Object);
      expect(typeof ImageMinimizerPlugin.loader).toBe("string");
      expect(typeof ImageMinimizerPlugin.loader).toBe("string");
      expect(typeof ImageMinimizerPlugin.imageminNormalizeConfig).toBe(
        "function"
      );
      expect(typeof ImageMinimizerPlugin.imageminMinify).toBe("function");
      expect(typeof ImageMinimizerPlugin.imageminGenerate).toBe("function");
      expect(typeof ImageMinimizerPlugin.squooshMinify).toBe("function");
      expect(typeof ImageMinimizerPlugin.squooshGenerate).toBe("function");
    });
  });

  describe("imageminNormalizeConfig", () => {
    it("should works", () => {
      expect(() =>
        ImageMinimizerPlugin.imageminNormalizeConfig({})
      ).toThrowErrorMatchingSnapshot();
      expect(() =>
        ImageMinimizerPlugin.imageminNormalizeConfig({ plugins: [] })
      ).toThrowErrorMatchingSnapshot();
      expect(() =>
        ImageMinimizerPlugin.imageminNormalizeConfig({ plugins: ["unknown"] })
      ).toThrowErrorMatchingSnapshot();
      expect(() =>
        ImageMinimizerPlugin.imageminNormalizeConfig({
          plugins: ["imagemin-unknown"],
        })
      ).toThrowErrorMatchingSnapshot();

      expect(
        ImageMinimizerPlugin.imageminNormalizeConfig({ plugins: ["mozjpeg"] })
      ).toMatchSnapshot();
      expect(
        ImageMinimizerPlugin.imageminNormalizeConfig({
          plugins: ["imagemin-mozjpeg"],
        })
      ).toMatchSnapshot();
      expect(
        ImageMinimizerPlugin.imageminNormalizeConfig({ plugins: [["mozjpeg"]] })
      ).toMatchSnapshot();
      expect(
        ImageMinimizerPlugin.imageminNormalizeConfig({
          plugins: [["mozjpeg", { quality: 0 }]],
        })
      ).toMatchSnapshot();
      expect(
        ImageMinimizerPlugin.imageminNormalizeConfig({}, {})
      ).toMatchSnapshot();
      expect(
        ImageMinimizerPlugin.imageminNormalizeConfig({ plugins: [true] }, {})
      ).toMatchSnapshot();
    });
  });
});
