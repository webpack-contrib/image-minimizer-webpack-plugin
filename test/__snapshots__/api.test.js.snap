// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api imageminNormalizeConfig should works 1`] = `"No plugins found for \`imagemin\`. Please read documentation."`;

exports[`api imageminNormalizeConfig should works 2`] = `"No plugins found for \`imagemin\`. Please read documentation."`;

exports[`api imageminNormalizeConfig should works 3`] = `
"Unknown plugin: imagemin-unknown

Did you forget to install the plugin?
You can install it with:

$ npm install imagemin-unknown --save-dev
$ yarn add imagemin-unknown --dev"
`;

exports[`api imageminNormalizeConfig should works 4`] = `
"Unknown plugin: imagemin-unknown

Did you forget to install the plugin?
You can install it with:

$ npm install imagemin-unknown --save-dev
$ yarn add imagemin-unknown --dev"
`;

exports[`api imageminNormalizeConfig should works 5`] = `
Object {
  "plugins": Array [
    [Function],
  ],
  "pluginsMeta": Array [
    Array [
      Object {
        "name": "imagemin-mozjpeg",
        "options": Object {},
        "version": "9.0.0",
      },
    ],
  ],
}
`;

exports[`api imageminNormalizeConfig should works 6`] = `
Object {
  "plugins": Array [
    [Function],
  ],
  "pluginsMeta": Array [
    Array [
      Object {
        "name": "imagemin-mozjpeg",
        "options": Object {},
        "version": "9.0.0",
      },
    ],
  ],
}
`;

exports[`api imageminNormalizeConfig should works 7`] = `
Object {
  "plugins": Array [
    [Function],
  ],
  "pluginsMeta": Array [
    Array [
      Object {
        "name": "imagemin-mozjpeg",
        "options": Object {},
        "version": "9.0.0",
      },
    ],
  ],
}
`;

exports[`api imageminNormalizeConfig should works 8`] = `
Object {
  "plugins": Array [
    [Function],
  ],
  "pluginsMeta": Array [
    Array [
      Object {
        "name": "imagemin-mozjpeg",
        "options": Object {
          "quality": 0,
        },
        "version": "9.0.0",
      },
    ],
  ],
}
`;

exports[`api imageminNormalizeConfig should works 9`] = `Object {}`;

exports[`api imageminNormalizeConfig should works 10`] = `
Object {
  "plugins": Array [],
  "pluginsMeta": Array [],
}
`;

exports[`api normalizeImageminConfig should works 1`] = `"No plugins found for \`imagemin\`. Please read documentation."`;

exports[`api normalizeImageminConfig should works 2`] = `"No plugins found for \`imagemin\`. Please read documentation."`;

exports[`api normalizeImageminConfig should works 3`] = `
"Unknown plugin: imagemin-unknown

Did you forget to install the plugin?
You can install it with:

$ npm install imagemin-unknown --save-dev
$ yarn add imagemin-unknown --dev"
`;

exports[`api normalizeImageminConfig should works 4`] = `
"Unknown plugin: imagemin-unknown

Did you forget to install the plugin?
You can install it with:

$ npm install imagemin-unknown --save-dev
$ yarn add imagemin-unknown --dev"
`;

exports[`api normalizeImageminConfig should works 5`] = `
Object {
  "plugins": Array [
    [Function],
  ],
  "pluginsMeta": Array [
    Array [
      Object {
        "name": "imagemin-mozjpeg",
        "options": Object {},
        "version": "9.0.0",
      },
    ],
  ],
}
`;

exports[`api normalizeImageminConfig should works 6`] = `
Object {
  "plugins": Array [
    [Function],
  ],
  "pluginsMeta": Array [
    Array [
      Object {
        "name": "imagemin-mozjpeg",
        "options": Object {},
        "version": "9.0.0",
      },
    ],
  ],
}
`;

exports[`api normalizeImageminConfig should works 7`] = `
Object {
  "plugins": Array [
    [Function],
  ],
  "pluginsMeta": Array [
    Array [
      Object {
        "name": "imagemin-mozjpeg",
        "options": Object {},
        "version": "9.0.0",
      },
    ],
  ],
}
`;

exports[`api normalizeImageminConfig should works 8`] = `
Object {
  "plugins": Array [
    [Function],
  ],
  "pluginsMeta": Array [
    Array [
      Object {
        "name": "imagemin-mozjpeg",
        "options": Object {
          "quality": 0,
        },
        "version": "9.0.0",
      },
    ],
  ],
}
`;

exports[`api normalizeImageminConfig should works 9`] = `Object {}`;

exports[`api normalizeImageminConfig should works 10`] = `
Object {
  "plugins": Array [],
  "pluginsMeta": Array [],
}
`;
